File: Programmers/스킬트리.swift
import Foundation

func solution(_ skill: String, _ skill_trees: [String]) -> Int {
    var answer = 0
    let skillArr = Array(skill)
    
    for tree in skill_trees {
        let arr = Array(tree) //[b, a, c, d, e]
        var str = ""
        arr.forEach { if skill.contains($0) { str += String($0) } }
        
        if str == skill { 
            answer += 1
        } else {
            let strArr = Array(str)
            var isRight = true
            for index in 0..<strArr.count {
                if strArr[index] != skillArr[index] { 
                    isRight = false 
                    break
                }
            }
            if isRight { answer += 1 }
        }
    }
    
    return answer
}

File: Programmers/스킬트리.cpp
#include <string>
#include <vector>

#include <iostream>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = skill_trees.size();
    // skill order array
    int arr_skill[26];
    for(int i = 0; i < 26; i++){
        // default : initialize to -1
        arr_skill[i] = -1;
    }
    // if letter is in string 'skill' change value to skill's order
    for(int pos = 0; pos < skill.length(); pos++){
        int letter = skill[pos];
        arr_skill[letter - 65] = pos;
    }
    
    // check skill tree
    for(int i = 0; i < skill_trees.size(); i++){
        // check current skill
        string curr = skill_trees[i];
        // compare string 'curr' with array
        int skillOrder = 0;
        for(int j = 0; j < curr.length(); j++){
            int letter = curr[j];
            if(arr_skill[letter - 65] > -1){
                if(arr_skill[letter - 65] > skillOrder){
                    answer--;
                    break;
                } else{
                    skillOrder++;
                }
            }
        }          
    }
    
    return answer;
}
